provides.py

```diff
@@ -14,16 +14,29 @@
 from charms.reactive import RelationBase
 from charms.reactive import hook
 from charms.reactive import scopes
+from charmhelpers.core.hookenv import is_leader
+from charmhelpers.core.hookenv import config
+from charmhelpers.core import unitdata
 
 
 class EtcdProvider(RelationBase):
     scope = scopes.GLOBAL
 
     @hook('{provides:etcd}-relation-{joined,changed}')
-    def changed(self):
-        pass
+    def joined_or_changed(self):
+        if is_leader():
+            cstring = self.connection_string()
+            self.set_remote(data={'connection_string': cstring})
 
-    @hook('{provides:etcd}-relation-{broken, departed}')
-    def broken(self):
-        self.remove_state('{relation_name}.available')
-        self.remove_state('{relation_name}.connected')
+    def connection_string(self):
+        if is_leader():
+            db = unitdata.kv()
+            cluster_data = db.get('etcd.cluster_data')
+            connection_string = ""
+            for u in cluster_data:
+                connection_string += ",http://{}:{}".format(cluster_data[u]['private_address'], config('port'))  # noqa
+            return connection_string.lstrip(',')
```

Comment:
Do i need to reimplement with the leadership layer or is `leadership.is_leader` a built in?